### PR TITLE
Use CDN for publisher list

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/3191

Browser PR: https://github.com/brave/brave-browser/pull/3242

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [x] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
- enable rewards via terminal
- enable rewards
- make sure that publisher list request connects to distro url with gzip header added

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: browser/net/brave_static_redirect_network_delegate_helper.cc [hunk 1]
```diff
@@ -50,9 +50,11 @@ int OnBeforeURLRequest_StaticRedirectWork(
     URLPattern(URLPattern::SCHEME_HTTPS, "https://ledger.mercury.basicattentiontoken.org/*"),
     URLPattern(URLPattern::SCHEME_HTTPS, "https://balance.mercury.basicattentiontoken.org/*"),
     URLPattern(URLPattern::SCHEME_HTTPS, "https://publishers.basicattentiontoken.org/*"),
+    URLPattern(URLPattern::SCHEME_HTTPS, "https://publishers-distro.basicattentiontoken.org/*"),
```
Comment (bridiver) [inline browser/net/brave_static_redirect_network_delegate_helper.cc]: can we alphabetize these?

Comment (NejcZdovc) [reply browser/net/brave_static_redirect_network_delegate_helper.cc]: so you want to mix staging and prod urls?

File: chromium_src/net/tools/transport_security_state_generator/input_file_parsers.cc [hunk 2]
```diff
@@ -315,7 +315,9 @@ bool ParseJSON(base::StringPiece json,
       { "name": "download.brave.com", "mode": "force-https", "policy": "custom", "pins": "brave"},
       { "name": "laptop-updates.brave.com", "mode": "force-https", "policy": "custom", "pins": "brave"},
       { "name": "sync.brave.com", "mode": "force-https", "policy": "custom", "pins": "brave"},
-      { "name": "sync-staging.brave.com", "mode": "force-https", "policy": "custom", "pins": "brave"}
+      { "name": "sync-staging.brave.com", "policy": "custom", "mode": "force-https", "pins": "brave"},
+      { "name": "publishers-distro.basicattentiontoken.org", "policy": "custom", "mode": "force-https", "pins": "brave"},
+      { "name": "publishers-staging-distro.basicattentiontoken.org", "policy": "custom", "mode": "force-https", "pins": "brave"}
```
Comment (diracdeltas) [inline chromium_src/net/tools/transport_security_state_generator/input_file_parsers.cc]: ++

**Comment by NejcZdovc** [comment]
@diracdeltas all good for this PR?

**Comment by diracdeltas** [comment]
@NejcZdovc yup

**Comment by NejcZdovc** [comment]
@diracdeltas can you please approve it

**Pull request merged**